lib/vtls/cipher_suite.c

```diff
@@ -163,6 +163,10 @@ enum {
   CS_TXT_LEN,
 };
 
+#ifdef _MSC_VER
+#pragma warning(push)
+#pragma warning(disable : 4310)
```

Comment:
Is there a universal fix or tweak here that we can apply to the code that fixes this warning?

Can we figure out which cast is applied to which constant?

Reply:
I understand the error:
(uint8_t) ((b) << 4 )
This is due after byte shift, the const number is larger then 8byte, then it get truncated.

Other solution is to enlarge the   uint8_t zip[6]; ->   uint16_t zip[6];

But I really don't know what this code is doing ...

Reply:
It cannot change data type, the whole point of this macro is to use bytes. It could possibly do an `& 0xff` to do the truncation explicitly.

Reply:
I will try your suggested fix.

Reply:
Great! your solution is working! 🎉